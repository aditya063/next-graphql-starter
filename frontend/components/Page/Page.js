import PropTypes from 'prop-types';
import { Query } from 'react-apollo';

import DisplayLoading from '../DisplayLoading/DisplayLoading';
import DisplayError from '../DisplayError/DisplayError';
import Layout from '../Layout/Layout';
import SignOn from '../SignOn/SignOn';
import { IS_LOGGED_IN } from '../../graphql/queries';

const Authenticated = React.memo(props => <Layout {...props} />);

const UnAuthenticated = React.memo(() => <SignOn />);

const Page = React.memo(props => (
  <Query query={IS_LOGGED_IN}>
    {({ loading, error, data }) => {
      if (loading) return <DisplayLoading />;
      if (error) return <DisplayError error={error} />;

      return data && data.isLoggedIn ? (
        <Authenticated {...props} />
      ) : (
        <UnAuthenticated {...props} />
      );
    }}
  </Query>
));

Authenticated.propTypes = {
  children: PropTypes.node.isRequired
};

Page.propTypes = {
  children: PropTypes.node.isRequired
};

export default Page;
