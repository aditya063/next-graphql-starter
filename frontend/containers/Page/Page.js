import PropTypes from 'prop-types';
import { Query } from 'react-apollo';

import DisplayLoading from '../../components/DisplayLoading/DisplayLoading';
import DisplayError from '../../components/DisplayError/DisplayError';
import SignOn from '../../components/SignOn/SignOn';
import { IS_LOGGED_IN } from '../../graphql/queries';
import * as sc from './Page.style';

const Authenticated = React.memo(props => props.children);

const UnAuthenticated = React.memo(() => <SignOn />);

const Page = React.memo(props => (
  <Query query={IS_LOGGED_IN}>
    {({ loading, error, data }) => {
      if (loading) return <DisplayLoading />;
      if (error) return <DisplayError error={error} />;

      return data.isLoggedIn ? (
        <Authenticated {...props} />
      ) : (
        <UnAuthenticated {...props} />
      );
    }}
  </Query>
));

Authenticated.propTypes = {
  children: PropTypes.node.isRequired
};

Page.propTypes = {
  children: PropTypes.node.isRequired
};

export default Page;
