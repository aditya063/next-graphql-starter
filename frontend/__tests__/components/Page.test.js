import { shallow } from 'enzyme';
import Page from '../../components/Page/Page';

describe('Page', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    mockProps = {
      children: [React.createElement('div', { key: 1 }, 'mock component')]
    };
    wrapper = shallow(<Page {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  afterEach(() => jest.resetAllMocks());

  // TODO: test loading and error states
  xit('matches snapshot - loading', () => {
    const wrapperLoading = wrapper
      .renderProp('children')({ loading: true })
      .dive();

    expect(wrapperLoading).toMatchSnapshot();
  });

  it('matches snapshot - UnAuthenticated', () => {
    const UnAuthenticated = wrapper
      .renderProp('children')({ data: { isLoggedIn: false } })
      .dive();

    expect(UnAuthenticated).toMatchSnapshot();
  });

  it('matches snapshot - Authenticated', () => {
    const Authenticated = wrapper
      .renderProp('children')({ data: { isLoggedIn: true } })
      .dive();

    expect(Authenticated).toMatchSnapshot();
  });
});
