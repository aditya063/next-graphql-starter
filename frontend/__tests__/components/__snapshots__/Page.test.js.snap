// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page matches snapshot - Authenticated 1`] = `
<Memo()>
  <div
    key="1"
  >
    mock component
  </div>
</Memo()>
`;

exports[`Page matches snapshot - UnAuthenticated 1`] = `
<Fragment>
  <SignIn />
  <SignUp />
</Fragment>
`;
